### PR TITLE
feat(templates): add support for stages

### PR DESCRIPTION
**Pull request by JordanSussman**
Add support for utilizing stages within templates.

xref: https://github.com/go-vela/community/issues/233

### CODE REVIEW

**Review by jbrockopp — DISMISSED** [review]
LGTM

**Review by ecrupper — DISMISSED** [review]
looks good to me — thanks for the PR!

**Review by jbrockopp — DISMISSED** [review]
LGTM

**Review by jbrockopp — DISMISSED** [review]
LGTM

File: compiler/native/compile_test.go [hunk 1]
```diff
@@ -1930,3 +1867,1124 @@ func Test_client_modifyConfig(t *testing.T) {
...
+			name: "golang",
+			args: args{
+				file:         "testdata/golang_inline_stages.yml",
+				pipelineType: "golang",
+				template:     false,
+				substitute:   false,
+			},
+			want: &yaml.Build{
+				Version: "1",
+				Metadata: *&yaml.Metadata{
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply compiler/native/compile_test.go]: fixed 👍

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by ecrupper — APPROVED** [review]
lgtm

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by wass3r — APPROVED** [review]
thanks 🥳

**Pull request merged**